script/parse-examples

```diff
@@ -3,28 +3,50 @@
 set -e
 
 cd "$(dirname "$0")/.."
+root="$PWD"
 
-function clone_repo {
-  owner=$1
-  name=$2
-  sha=$3
+clone_repo() {
+  local owner=$1
+  local name=$2
+  local sha=$3
+  local folder="$4"
+  local path="$root/examples/$name"
 
-  path=examples/$name
-  if [ ! -d "$path" ]; then
+  if [ -d "$path" ]; then
+    pushd "$path" > /dev/null
+    if [ "$(git rev-parse HEAD 2>/dev/null)" == "$sha"  ]; then
+      popd > /dev/null
+      return
+    else
+      popd > /dev/null
+      rm -rf "$path"
+      echo "Updating $owner/$name to $sha"
+    fi
+  else
     echo "Cloning $owner/$name"
-    git clone "https://github.com/$owner/$name" "$path"
   fi
 
+  mkdir -p "$path"
   pushd "$path" > /dev/null
-  if [ "$(git rev-parse head)" != "$sha"  ]; then
-    echo "Updating $owner/$name to $sha"
-    git fetch
-    git reset --hard $sha
+  git init
+  git remote add origin "https://github.com/$owner/$name"
+  git pull --ff-only --depth 1 origin "$sha"
+  if [ "${folder:-}" ]; then
+    while IFS= read -r file; do
+      echo "${folder}" "$file"
+      if [ "$file" != "$folder" ]; then
+        rm -r "$file"
```

Comment:
This should be `rm -rf` instead of `-r` only because the `ls -A` will expose `.git` and other hidden files which might be write-protected. Otherwise, you'll get prompts like those:

```sh
> rm -r .git
rm: remove write-protected regular file '.git/objects/58/7ff082e0b98914788500eae5dd6a33f04883c9'? 
```